### DISCUSSION TITLE
Modules best practices

### DISCUSSION THREAD

**Discussion by divan** [geldata/gel#5221] [General]
What would be the best practices for using modules?

I have couple of projects where I put evertything in default module and it's okay, but my business logic code is grouped into logical parts that are only loosely connected to each other. Would it be a good practice to move objects in schema to different modules to give more clarity to the structure? Is there any downsides for this?

I've read in some comments that there were plans to support "installing" modules, which is interesting idea, but I can't find any good explanation of the initial logic behind modules and their intented use, including best real-world practices. Any advice what to read?

**Comment by tailhook** [comment]
The only nuance is that you have to specify namespace for non-default namespace:
1. In queries, if you're not using query builder (probably using `with module <your-module-name>`)
2. In other modules (which is often a good thing not a downside)

You can also use `with_default_module` in connection pool for (1), but that has to be used carefully (for example good where each microservice have their own primary module, but may be messy within monolithic app when different application modules can call each other)

-----

Generally I'd say that modules should be used as in any other programming language at your discretion. Probably making the modules too fine-grained will be annoying, but everyone have their own understanding of how small or how big modules should be.
